test/ecto/adapters/mysql_test.exs

```diff
@@ -222,6 +232,19 @@ defmodule Ecto.Adapters.MySQLTest do
     assert all(query) == ~s{SELECT NOT (s0.`x` IS NULL) FROM `schema` AS s0}
   end
 
+  test "order_by and types" do
+    query =
+      normalize from(e in "schema4",
+        order_by: type(fragment("(?->>?)", field(e, ^:map), ^"value"), ^:decimal),
+        select: true)
+
+    result =
+      "SELECT TRUE FROM `schema4` AS s0 " <>
+      "ORDER BY (s0.`map`->>?) + 0"
+
+    assert all(query) == String.trim(result)
```

Comment:
You don't need trim here. YOu can keep everything in a single line:

    assert all(query) ==  "SELECT TRUE FROM \"schema4\" AS s0 ORDER BY (s0.\"map\"->>$1)::decimal"